app/streaming/video/videoenhancement.cpp

```diff
@@ -0,0 +1,318 @@
+#include "videoenhancement.h"
+#include <QtDebug>
+#include <regex>
+
+#ifdef Q_OS_WIN
+#include <windows.h>
+#include <dxgi.h>
+#include <d3d11.h>
+#include <wrl/client.h>
+#pragma comment(lib, "Advapi32.lib")
+#endif
+
+/**
+ * \brief Constructor (Singleton)
+ *
+ * Check the capacity to handle the AI-Enhancement features such as Video Super-Resolution or SDR to HDR, according to multiple parameters such as OS or Video driver.
+ *
+ * \return void
+ */
+VideoEnhancement::VideoEnhancement()
+{
+    if(!m_Initialized){
+        setGPUinformation();
+        // Avoid to set variables every call of the instance
+        m_Initialized = true;
+    }
+}
+
+/**
+ * \brief Get the singleton instance
+ *
+ * Render the instance of the singleton
+ *
+ * \return VideoEnhancement instance
+ */
+VideoEnhancement &VideoEnhancement::getInstance()
+{
+    static VideoEnhancement instance;
+    return instance;
+}
+
+/**
+ * \brief Retreive GPU information
+ *
+ * Retreive all GPU information: Vendor ID, Driver version, GPU name
+ *
+ * \return bool Returns true if it successfully retreived the GPU information
+ */
+bool VideoEnhancement::setGPUinformation()
+{
+    bool success = false;
+
+#ifdef Q_OS_WIN
+
+    // Create a Direct3D 11 device
+    ID3D11Device* pD3DDevice = nullptr;
+    ID3D11DeviceContext* pContext = nullptr;
+
+    HRESULT hr = D3D11CreateDevice(
```

Comment:
Done
It will select the most appropriate GPU based on VSR and HDR capabilities and quality (our own perception) per Vendor.
It will keep the previous behavior (adapter linked to display) if Video Enhancement is disabled.